src/main/java/io/stargate/sgv2/jsonapi/api/v1/metrics/JsonShreddingMetricsReporter.java

```diff
@@ -0,0 +1,39 @@
+package io.stargate.sgv2.jsonapi.api.v1.metrics;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.core.instrument.Timer;
+
+/** Reports the time and size of json shredding. */
+public class JsonShreddingMetricsReporter {
+
+  private final MeterRegistry meterRegistry;
+  private final JsonApiMetricsConfig jsonApiMetricsConfig;
+
+  private Timer.Sample sample;
+
+  public JsonShreddingMetricsReporter(
+      MeterRegistry meterRegistry, JsonApiMetricsConfig jsonApiMetricsConfig) {
+    this.meterRegistry = meterRegistry;
+    this.jsonApiMetricsConfig = jsonApiMetricsConfig;
+  }
+
+  public void startMetrics() {
+    sample = Timer.start(meterRegistry);
+  }
+
+  public void completeMetrics(int totalJsonStringLen, int numberOfDocs, String commandName) {
```

Comment:
Since we can now get multiple documents, should probably have

    long totalJsonStringLen

instead of `int`.